python/paddle/fluid/layers/nn.py

```diff
@@ -4710,6 +4711,73 @@ def multiplex(inputs, index):
     return out
 
 
+def mask_extract(input, mask, name=None):
```

Comment:
this API looks so much like `gather`, with a little bit different on how you prepare `index` or `mask`. Can you just use gather?

It's also a bit like look_up.

Or, you can add/find an op to convert `mask` to `index`?